authenticator.py

```diff
@@ -74,10 +78,18 @@ def authenticate(cls, request):
             get_nio_logger("Basic.Authenticator").error(msg)
             raise Unauthorized(msg)
 
-        # check password
-        if base64_encode(password) != user.get('password'):
-            msg = "Password is invalid."
-            get_nio_logger("Basic.Authenticator").error(msg)
-            raise Unauthorized(msg)
-
-        return User(name=username)
+        stored_pwd = user.get('password')
+        # validate bcrypt hashed password
+        if re.search('^\$2.{1}\$.{56}$', stored_pwd):
```

Comment:
This is a typical regex match for a bcrypt hash. I substituted something like `[ayb]` for just a general one character match if future bcrypt patches are released.